components/event-publisher/event-output-adapters/org.wso2.carbon.event.output.adapter.email/src/main/java/org/wso2/carbon/event/output/adapter/email/EmailEventAdapter.java

```diff
@@ -232,22 +243,24 @@ public void connect() throws ConnectionUnavailableException {
             try {
                 smtpFromAddress = new InternetAddress(smtpFrom);
             } catch (AddressException e) {
-                String msg = "failed to connect to the mail server due to error in retrieving " +
-                        "smtp from address";
+                String msg = "failed to connect to the mail server due to error in retrieving smtp from address";
                 if (log.isDebugEnabled()) {
                     log.debug(msg + ": " + smtpFrom);
                 }
                 throw new ConnectionUnavailableException
                         ("The adapter " + eventAdapterConfiguration.getName() + " " + msg, e);
             }
 
-            //Retrieving username and password of SMTP server.
-            smtpUsername = props.getProperty(MailConstants.MAIL_SMTP_USERNAME);
-            smtpPassword = props.getProperty(MailConstants.MAIL_SMTP_PASSWORD);
-
+            Map<String, String> credentials = resolveCredentials(globalProperties);
+            smtpUsername = credentials.get(USERNAME);
+            smtpPassword = credentials.get(PASSWORD);
 
             //initializing SMTP server to create session object.
             if (smtpUsername != null && smtpPassword != null && !smtpUsername.isEmpty() && !smtpPassword.isEmpty()) {
+                if (CLIENT_CREDENTIAL.equalsIgnoreCase(props.getProperty(ADAPTER_EMAIL_AUTH_TYPE))) {
+                    props.put(ADAPTER_EMAIL_SMTP_AUTH_MECHANISMS, XOAUTH2);
```

Comment:
What is the meaning of "XOAUTH2"?

Reply:
XOAUTH2 is a protocol used for authentication. Instead of using a traditional username and password, XOAUTH2 uses an OAuth 2.0 access token for authentication.